rtengine/image16.cc

```diff
@@ -284,6 +284,46 @@ void Image16::getStdImage (ColorTemp ctemp, int tran, Imagefloat* image, Preview
 #undef GCLIP
 }
 
+bool
+Image16::isBW() const
+{
+    if (!height || !width) {
+        return false;
+    }
+
+    bool res = true;
+
+    for ( int w = 0, h = height / 2; res && w < width; ++w ) {
+        res =
+            res
+            && (
+                r(h, w) == g(h, w)
+                || r(h, w) == b(h, w)
```

Comment:
I tested an [arithmetic version of the loop](https://gist.github.com/heckflosse/efb2b3cb5ac229764618a23a2040cf0b). It's about 12 % faster, but as we are in 16 bit integer space here it's already very fast in original version. On my machine for a full size 50 Mp output it needs 32 ms in original version and 28 ms in arithmetic version.